File: backend/models/car.py
from . import db  

class Car(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(255), nullable=False)
    year = db.Column(db.Integer)
    model = db.Column(db.String(255))
    brand = db.Column(db.String(255))
    km = db.Column(db.Integer)
    type = db.Column(db.String(50))
    code = db.Column(db.String(50))
    engine = db.Column(db.String(255))  
    transmission = db.Column(db.String(50)) 
    horsepower = db.Column(db.Integer)  
    topSpeed = db.Column(db.Integer) 
    availability = db.Column(db.Boolean, default=True)
    description = db.Column(db.Text)
    picture = db.Column(db.String(255))  #

File: frontend/src/Components/types/car.ts
export interface Car {
  id: number;
  name: string;
  brand: string;
  model: string;
  year: number;
  price: number;
  picture: string;
  km? : number;
  type?: string;
  code?: string;
  engine?: string;
  transmission?: string;
  horsepower?: number;
  topSpeed?: number;
  availability?: boolean;
  description?: string;
}

export interface CarDetails extends Car {
  description: string;
  specs: {
    engine: string;
    transmission: string;
    horsepower: number;
    acceleration: string;
    topSpeed: string;
  };
}
